free-3.md

```diff
@@ -10,34 +10,48 @@ _What need is met by your idea? Who are you helping? What is that person's
+
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+A DSL is appropriate because poems, and especially structured poems, are a very particular and strictly defined subset of natural language, that can be checked as long as the rules are properly described. I think it addresses the need by acting as a domain-specific spellchecker, to which inputs (which type of poem, the poem itself) can be input, and suggestions can be output.
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+I like the idea because it seems like something that could be done in a fairly straightforward way. I came up with it just thinking of DSL ideas -- it seemed like something I could do and something that would be fun to play around with.
 
 ### Domain
 _Describe the project's domain in five words._
 
+Poems with well defined structure
+
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+I think the user would input to the language a specific poem type, as well as a text file or string which is the poem itself. I think a text file would be logical because then the program could look for line breaks more easily. The program would then output either a list of suggestions referencing line numbers and specific words, or maybe a text file of suggestions based on the poem type, or some similar format.
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+Once the program receives the desired inputs from the user, it would cross-check the poem type against existing poem types that it has. Then, it would look up each word individually against a dictionary (either something stored to a database or using Webster or something through an API). It would check :
+	- that the word exists or is a proper noun
+	- that the word is in the appropriate rhyme scheme
+	- that meter (i.e. iambic pentameter) is observed on a word and line level
+
+Errors would occur when poem types that don't exist are attempted, or perhaps when weird symbols are entered in a poem (though this should really be something that can be caught by the program).
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+It should be easy to see that poems are correctly formatted, provided the poem type and a poem. It should be possible but difficult to write non-poems or prose and run it through the checker. It should be impossible to do things like ask the language if a poem is good or something like that.
```

Comment:
Although this might be going out of scope, outputting a formatted PDF or LaTeX would really cool.